lib/active_admin/resource_collection.rb

```diff
@@ -19,6 +19,10 @@ def add(resource)
       end
     end
 
+    def delete(resource)
+      @collection.reject! { |k| k == resource.resource_name }
```

Comment:
Absolutely!

However, this is the only change affecting the actual code in `lib` and it's not really in the scope of this PR. I only added it while debugging some failures but it's not really necessary. So I'd say it's best to just remove it.